### PR TITLE
Show warnings returned in admission reviews

### LINKED ISSUE #1474: Show warnings returned in admission reviews when updating shoots
**What would you like to be added**:
The dashboard provides the means to edit a shoot spec directly from the browser. If an admission webhook denies the request the returned error is displayed to the user, however if the request is allowed and returned with a warning that warning is not displayed to the user. It will be nice if such warnings are shown in the dashboard as well. Similarly if I edit a resource via `kubectl` and a warning is returned that is displayed in the terminal.

```bash
k -n garden-local edit shoot local
Warning: my cool warning message
shoot.core.gardener.cloud/local edited
```

**Why is this needed**:
This will improve user experience in a sense that it might bring attention to potential problems and misconfigurations.

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:
- Show kubernetes warnings as alert
- Make update errors persistent alerts
- Refactored app store to use notify composable

Screenshot 2024-10-08 at 17 38 36

**Which issue(s) this PR fixes**:
Fixes #1474

**Special notes for your reviewer**:
- @holgerkoser Check changes to kubernetes client, Also currently only create / update methods support to return response
- Could also add logic for edit workes as there also the yaml can be edited WDYT?

**Release note**:

```feature user
Warnings returned in admission reviews are now displayed when creating or updating clusters using the cluster editor
```

### CODE REVIEW

File: packages/request/lib/Client.js [hunk 1]
```diff
@@ -320,6 +320,11 @@ class Client {
         body
       })
     }
+
+    if (returnResponse) {
```
Comment (petersutter) [inline packages/request/lib/Client.js:324]: as you only need the headers for this feature, how about making use of the afterResponse hook (line 314)?

Comment (grolu) [reply packages/request/lib/Client.js:337]: I think the headers are not available in the hook but I'm not 100% sure. I remember that I tried to use the hook but it was not possible. @holgerkoser may has a better idea regarding the implementation

Comment (holgerkoser) [reply packages/request/lib/Client.js]: I have added an option `onWarning` to the kube-client that give access to the warning header value without changing the method signatures.

File: frontend/src/store/app.js [hunk 2]
```diff
@@ -51,9 +51,16 @@ export const useAppStore = defineStore('app', () => {
     } else {
       alert.message = ''
     }
+    if (persistent) {
+      alert.duration = -1
+    }
     setAlert(alert)
   }
 
+  function setPersistentError (value) {
```
Comment (holgerkoser) [inline frontend/src/store/app.js:60]: Do not add new methods. The existing api can handle this case.
```js
  function setAlertWithType (type, value) {
    const alert = {
      type,
    }
    if (typeof value === 'string') {
      alert.message = value
    } else if (value) {
      const { message = '', title, duration } = value
      alert.message = value.response
        ? errorDetailsFromError(value).detailedMessage
        : message
      if (title) {
        alert.title = title
      }
      if (duration) {
        alert.duration= duration
      }
    } else {
      alert.message = ''
    }
    setAlert(alert)
  }
```
Usage:
```js
  appStore.setError({
    message: `Failed to reset Service Account: ${err.message}`,
    duration: -1
  })
```

File: frontend/src/views/GNewShoot.vue [hunk 3]
```diff
@@ -243,11 +244,15 @@ export default {
...
 
       try {
-        await this.api.createShoot({
+        const { headers } = await this.api.createShoot({
           namespace: this.shootNamespace,
           data: this.shootManifest,
         })
-        this.setSuccess('Cluster created')
+        if (headers.warning) {
+          this.setPersistentWarning({ message: headers.warning })
```
Comment (holgerkoser) [inline frontend/src/views/GNewShoot.vue:252]: The text is not renderd correctly. Long strings are cut off. We could use https://github.com/marcbachmann/warning-header-parser or create our own implementation.

Comment (grolu) [reply frontend/src/views/GNewShoot.vue]: Added parser

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Pull request merged**